app/components/developer.py

```diff
@@ -39,10 +56,185 @@ async def status(self, interaction: dc.Interaction) -> None:
             await interaction.response.send_message(
                 "Only mods can use this command.", ephemeral=True
             )
+            return
         await interaction.response.send_message(
             await self.bot.bot_status.status_message(), ephemeral=True
         )
 
+    @dc.app_commands.command(description="Reload bot extensions.")
+    @dc.app_commands.guild_only()
+    # Hide interaction from non-mods
+    @dc.app_commands.default_permissions(ban_members=True)
+    @dc.app_commands.autocomplete(extension=existing_extension_autocomplete)
+    async def reload(
+        self, interaction: dc.Interaction, extension: str | None = None
+    ) -> None:
+        # The client-side check with `default_permissions` isn't guaranteed to work.
+        if not self.bot.is_ghostty_mod(interaction.user):
+            await interaction.response.send_message(
+                "Only mods can use this command.", ephemeral=True
+            )
+            return
+
+        if extension:
+            if not self.bot.is_valid_extension(extension):
+                await interaction.response.send_message(
+                    f"Extension `{extension}` does not exist or is invalid.",
+                    ephemeral=True,
+                )
+                return
+            extensions = [extension]
+        else:
+            # If no extension is provided, reload all extensions
+            extensions = self.bot.get_component_extension_names()
+
+        reloaded_extensions: list[str] = []
+        failed_reloaded_extensions: list[str] = []
+
+        await interaction.response.defer(thinking=True, ephemeral=True)
+        for ext in extensions:
+            try:
+                try:
+                    await self.bot.unload_extension(ext)
+                except commands.ExtensionNotLoaded:
+                    # If already not loaded, load it
+                    await self.bot.load_extension(ext)
+                else:
+                    await self.bot.load_extension(ext)
```

Comment:
How do you feel about this?
```suggestion
                with suppress(commands.ExtensionNotLoaded):
                    await self.bot.unload_extension(ext)
                await self.bot.load_extension(ext)
```